### ISSUE TITLE
NullPointerException when asserting list containing null

### ISSUE THREAD

**Issue by francisdb** [etorreborre/specs2#350]
```scala
List(TRUE, TRUE) is not equal to List(TRUE, null)
```

```scala
[error]   x reset observations
[error]    List(TRUE, TRUE) is not equal to List(TRUE, null) (PeriodicScenarioSpec.scala:91)
[error]
[error] java.lang.NullPointerException
[error]
[error] STACKTRACE
[error]   org.specs2.main.SmartDiffs$$anonfun$showSeqDiffs$2.apply(Diffs.scala:72)
[error]   org.specs2.main.SmartDiffs$$anonfun$showSeqDiffs$2.apply(Diffs.scala:72)
[error]   scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
[error]   scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
[error]   scala.collection.immutable.List.foreach(List.scala:381)
[error]   scala.collection.TraversableLike$class.map(TraversableLike.scala:245)
[error]   scala.collection.immutable.List.map(List.scala:285)
[error]   org.specs2.main.SmartDiffs.showSeqDiffs(Diffs.scala:72)
[error]   org.specs2.reporter.TextPrinter$$anonfun$printFailureDetails$1.apply(TextPrinter.scala:224)
[error]   org.specs2.reporter.TextPrinter$$anonfun$printFailureDetails$1.apply(TextPrinter.scala:212)
[error]   org.specs2.reporter.TextPrinter$$anonfun$printFailure$3.apply(TextPrinter.scala:149)
[error]   org.specs2.reporter.TextPrinter$$anonfun$printFailure$3.apply(TextPrinter.scala:149)
[error]   scalaz.stream.Process$$anonfun$append$1.apply(Process.scala:60)
[error]   scalaz.stream.Process$$anonfun$append$1.apply(Process.scala:59)
[error]   scalaz.stream.Process$$anonfun$2$$anonfun$apply$12$$anonfun$apply$13.apply(Process.scala:114)
[error]   scalaz.stream.Process$$anonfun$2$$anonfun$apply$12$$anonfun$apply$13.apply(Process.scala:114)
[error]   scalaz.stream.Util$.Try(Util.scala:38)
[error]   scalaz.stream.Process$$anonfun$2$$anonfun$apply$12.apply(Process.scala:114)
[error]   scalaz.stream.Process$$anonfun$2$$anonfun$apply$12.apply(Process.scala:114)
[error]   scalaz.Trampoline$$anonfun$delay$1.apply(Free.scala:234)
[error]   scalaz.Trampoline$$anonfun$delay$1.apply(Free.scala:234)
[error]   scalaz.std.FunctionInstances$$anon$1$$anonfun$map$1.apply(Function.scala:56)
[error]   scalaz.Free$$anonfun$run$1.apply(Free.scala:172)
[error]   scalaz.Free$$anonfun$run$1.apply(Free.scala:172)
[error]   scalaz.Free.go2$1(Free.scala:119)
[error]   scalaz.Free.go(Free.scala:122)
[error]   scalaz.Free.run(Free.scala:172)
[error]   scalaz.stream.Process$$anonfun$go$1$1.apply(Process.scala:83)
[error]   scalaz.stream.Process$$anonfun$go$1$1.apply(Process.scala:83)
[error]   scalaz.stream.Util$.Try(Util.scala:38)
[error]   scalaz.stream.Process$class.go$1(Process.scala:83)
[error]   scalaz.stream.Process$class.step(Process.scala:96)
[error]   scalaz.stream.Process$Append.step(Process.scala:616)
[error]   scalaz.stream.Process$class.go$3(Process.scala:469)
[error]   scalaz.stream.Process$$anonfun$go$3$2.apply(Process.scala:474)
[error]   scalaz.concurrent.Task$$anonfun$flatMap$1$$anonfun$1.apply(Task.scala:36)
[error]   scalaz.concurrent.Task$$anonfun$flatMap$1$$anonfun$1.apply(Task.scala:36)
[error]   scalaz.concurrent.Task$.Try(Task.scala:385)
[error]   scalaz.concurrent.Task$$anonfun$flatMap$1.apply(Task.scala:36)
[error]   scalaz.concurrent.Task$$anonfun$flatMap$1.apply(Task.scala:34)
[error]   scala.Function1$$anonfun$andThen$1.apply(Function1.scala:55)
[error]   scala.Function1$$anonfun$andThen$1.apply(Function1.scala:55)
[error]   scalaz.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:59)
[error]   scalaz.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:59)
[error]   scalaz.concurrent.Future.step(Future.scala:110)
[error]   scalaz.concurrent.Future.listen(Future.scala:76)
[error]   scalaz.concurrent.Future$$anonfun$listen$1$$anonfun$apply$4.apply(Future.scala:80)
[error]   scalaz.concurrent.Future$$anonfun$listen$1$$anonfun$apply$4.apply(Future.scala:80)
[error]   scalaz.Free$$anonfun$map$1.apply(Free.scala:52)
[error]   scalaz.Free$$anonfun$map$1.apply(Free.scala:52)
[error]   scalaz.Free.resume(Free.scala:73)
[error]   scalaz.Free.go2$1(Free.scala:118)
[error]   scalaz.Free.go(Free.scala:122)
[error]   scalaz.Free.run(Free.scala:172)
[error]   scalaz.concurrent.Future.listen(Future.scala:77)
[error]   scalaz.concurrent.Future$$anon$1$$anonfun$reduceUnordered$2$$anonfun$apply$10.apply(Future.scala:310)
[error]   scalaz.concurrent.Future$$anon$1$$anonfun$reduceUnordered$2$$anonfun$apply$10.apply(Future.scala:309)
[error]   scala.collection.Iterator$class.foreach(Iterator.scala:750)
[error]   scala.collection.AbstractIterator.foreach(Iterator.scala:1202)
[error]   scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
[error]   scala.collection.AbstractIterable.foreach(Iterable.scala:54)
[error]   scalaz.concurrent.Future$$anon$1$$anonfun$reduceUnordered$2.apply(Future.scala:309)
[error]   scalaz.concurrent.Future$$anon$1$$anonfun$reduceUnordered$2.apply(Future.scala:305)
[error]   scalaz.concurrent.Future.listen(Future.scala:80)
[error]   scalaz.concurrent.Future.runAsync(Future.scala:143)
[error]   scalaz.concurrent.Future.run(Future.scala:160)
[error]   scalaz.concurrent.Task.attemptRun(Task.scala:104)
[error]   org.specs2.control.ActionT$.fromTask(ActionT.scala:125)
[error]   org.specs2.control.ActionTSupport$class.fromTask(ActionT.scala:188)
[error]   org.specs2.control.package$Actions$.fromTask(package.scala:29)
[error]   org.specs2.reporter.Reporter$$anonfun$report$1.apply(Reporter.scala:47)
[error]   org.specs2.reporter.Reporter$$anonfun$report$1.apply(Reporter.scala:37)
[error]   org.specs2.runner.SbtRunner$$anonfun$specificationRun$1$$anonfun$3.apply(SbtRunner.scala:75)
[error]   org.specs2.runner.SbtRunner$$anonfun$specificationRun$1$$anonfun$3.apply(SbtRunner.scala:75)
[error]   org.specs2.control.ActionT$$anonfun$flatMap$1$$anonfun$apply$2.apply(ActionT.scala:36)
[error]   org.specs2.control.ActionT$$anonfun$flatMap$1$$anonfun$apply$2.apply(ActionT.scala:36)
[error]   org.specs2.control.StatusT$$anonfun$flatMap$1.apply(StatusT.scala:20)
[error]   org.specs2.control.StatusT$$anonfun$flatMap$1.apply(StatusT.scala:19)
[error]   scala.Function1$$anonfun$andThen$1.apply(Function1.scala:55)
[error]   scalaz.WriterT$$anonfun$flatMapF$1.apply(WriterT.scala:63)
[error]   scalaz.WriterT$$anonfun$flatMapF$1.apply(WriterT.scala:62)
[error]   scalaz.effect.IO$$anonfun$flatMap$1$$anonfun$apply$9.apply(IO.scala:62)
[error]   scalaz.effect.IO$$anonfun$flatMap$1$$anonfun$apply$9.apply(IO.scala:61)
[error]   scalaz.Free$$anonfun$flatMap$1$$anonfun$apply$1.apply(Free.scala:60)
[error]   scalaz.Free$$anonfun$flatMap$1$$anonfun$apply$1.apply(Free.scala:60)
[error]   scalaz.Free.resume(Free.scala:72)
[error]   scalaz.Free.go2$1(Free.scala:118)
[error]   scalaz.Free.go(Free.scala:122)
[error]   scalaz.Free.run(Free.scala:172)
[error]   scalaz.effect.IO$class.unsafePerformIO(IO.scala:22)
[error]   scalaz.effect.IOFunctions$$anon$6.unsafePerformIO(IO.scala:227)
[error]   org.specs2.runner.SbtRunner$$anonfun$newTask$1$$anon$4.execute(SbtRunner.scala:40)
[error]   sbt.ForkMain$Run$2.call(ForkMain.java:294)
[error]   sbt.ForkMain$Run$2.call(ForkMain.java:284)
[error]   java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error]   java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[error]   java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[error]   java.lang.Thread.run(Thread.java:745)
[error]
[error]
[error] This looks like a specs2 exception...
[error] Please report it with the preceding stacktrace at http://github.com/etorreborre/specs2/issues
```

**Comment by etorreborre** [comment]
`null` are such a big hole in the type system... This is fixed in `3.2-20150329233502-0653950`.

**Comment by francisdb** [comment]
True. Testing legacy java code...
